service/storage_service.cc

```diff
@@ -291,11 +291,11 @@ static future<> set_gossip_tokens(gms::gossiper& g,
     assert(!tokens.empty());
 
     // Order is important: both the CDC streams timestamp and tokens must be known when a node handles our status.
-    return g.add_local_application_state({
-        { gms::application_state::TOKENS, gms::versioned_value::tokens(tokens) },
-        { gms::application_state::CDC_GENERATION_ID, gms::versioned_value::cdc_generation_id(cdc_gen_id) },
-        { gms::application_state::STATUS, gms::versioned_value::normal(tokens) }
-    });
+    return g.add_local_application_state(
+        std::pair(gms::application_state::TOKENS, gms::versioned_value::tokens(tokens)),
+        std::pair(gms::application_state::CDC_GENERATION_ID, gms::versioned_value::cdc_generation_id(cdc_gen_id)),
+        std::pair(gms::application_state::STATUS, gms::versioned_value::normal(tokens))
+    );
```

Comment:
It's a bit sad that we have to type the `std::pair` now instead of using the braces, but it looks like template argument deduction doesn't work well with brace initializer lists, so I don't have a good suggestion. Not a big issue, just slightly more boilerplate.

Reply:
Yes I was sad to have to add it too.